sql/driver/wire.go

```diff
@@ -80,14 +80,15 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = t.DateVal.GoUTCTime()
 	case *Datum_TimeVal:
-		val = t.TimeVal.GoTime().UTC()
+		val = t.TimeVal.GoUTCTime()
+	case *Datum_IntervalVal:
+		val = time.Duration(t.IntervalVal)
 	}
 
-	if driver.IsValue(val) {
-		return val, nil
-	}
-	return nil, util.Errorf("unsupported type %T", val)
```

Comment:
Because time.Duration is indeed an unsupported type. See the commit message: we are deliberately violating the `driver.Value` contract here, but it seems to work.